src/geo_traits_impl/line_string.rs

```diff
@@ -0,0 +1,59 @@
+use super::{LineStringType, PointType};
+use bytemuck::TransparentWrapper;
+use geo_traits::{Dimensions, LineStringTrait, PointTrait};
+
+impl LineStringTrait for LineStringType {
+    type T = f64;
+    type CoordType<'b>
+        = &'b PointType
+    where
+        Self: 'b;
+
+    fn num_coords(&self) -> usize {
+        self.0.len()
+    }
+
+    fn coord(&self, i: usize) -> Option<Self::CoordType<'_>> {
+        Some(PointType::wrap_ref(&self.0[i]))
+    }
+
+    unsafe fn coord_unchecked(&self, i: usize) -> Self::CoordType<'_> {
+        self.coord(i).unwrap()
+    }
+
+    fn coords(&self) -> impl DoubleEndedIterator + ExactSizeIterator<Item = Self::CoordType<'_>> {
+        self.0.iter().map(PointType::wrap_ref)
+    }
+
+    fn dim(&self) -> Dimensions {
+        self.coord(0).unwrap().dim() // TODO: is this okay?
```

Comment:
TODO to remove / resolve

Reply:
An empty linestring is a normal occurrence, so I don't think we should be `unwrap` here.

My guess would be something like `Dimensions::Unknown(0)` or just `Dimensions::Xy` since it's usually correct.

Reply:
Perhaps we should document in geo-traits the expected values of `Dimensions::Unknown` and whether `0` should be allowed or not.

Reply:
We face a similar issue in WKT: https://github.com/georust/wkt/issues/125

WKT seemingly has a solution, but it doesn't really apply to geojson, because the dimensions of geojson are only ever implicit.

Reply:
I'd say it's not really an issue in WKT because the WKT string does explicitly say what dimension it is. Here it truly is implicit.

Reply:
Addressed in https://github.com/georust/geojson/pull/245/commits/d90e8765c7de93fdf4c8425a4681c6b306990cea